### PR TITLE
Helpers and fixes

### PR DESCRIPTION
**Pull request by dNechita**
Adding a couple of helpers that will be required for the implementation of the new plugin which will support multiple adrv9009 devices.
I've also squeezed a couple of small fixes.

### CODE REVIEW

File: iio_widget.c [hunk 1]
```diff
@@ -77,6 +77,8 @@ static void iio_widget_init(struct iio_widget *widget,
 	widget->update_value = update_value;
 	widget->save = save;
 	widget->priv = priv;
+	widget->priv_progress = NULL;
+	widget->priv_convert_function = NULL;
```
Comment (commodo) [inline iio_widget.c]: would a `memset(widget, 0, sizeof(*widget))` work instead at the beginning of this init block ?
the added benefit is that if you extend the `iio_widget` struct later, you can be sure that the new fields are 0/NULL if un-initialized explicitly;

File: libini2.h [hunk 2]
```diff
@@ -20,4 +20,6 @@ int foreach_in_ini(const char *ini_file,
 
 int ini_unroll(const char *input, const char *output);
 
+void write_driver_name_to_ini(FILE *f, const char *driver_name);
```
Comment (commodo) [inline libini2.h]: curios: will this be used later ?
because if it won't be, then maybe we could just leave it as static inside `libini2.c`

**Comment by commodo** [comment]
Looks good.

I'd add some descriptions to the commits to detail why they're being done.
It's not quite clear what these functions will be used for, or why they are being added or exported.

Alternatively, they can be added/exported in the same commit that uses them; which means grouping these patches with commits that use each function that is added/exported.
Either way works [i.e. commit description detailing their future role [or] grouping these commits with commits that will use these functions].

File: libini2.h (hunk 2 above)
Comment (dNechita) [reply libini2.h]: Yes, it will be used later. I've mentioned this in the PR description.

File: iio_widget.c (hunk 1 above)
Comment (dNechita) [reply iio_widget.c]: Yes, that will be better.

**Comment by dNechita** [comment]
I've added the requested comments.

**Comment by commodo** [comment]
[LGTM](https://scontent.fdub2-2.fna.fbcdn.net/v/t1.0-9/50589056_2067228833365893_7527438199266738176_n.png?_nc_cat=107&_nc_ht=scontent.fdub2-2.fna&oh=c8fbd979dd2b5188e8161fb07f1ebe4f&oe=5CFD7462)

**Comment by commodo** [comment]
Will merge tomorrow if no objections

**Pull request merged**